pkg/rest/extension_upload.go

```diff
@@ -35,7 +35,11 @@ func uploadExtension(client *http.Client, apiPath string, extensionName string,
 		}, err
 	}
 
-	resp := postMultiPartFile(client, apiPath, buffer, contentType, apiToken)
+	resp, err := postMultiPartFile(client, apiPath, buffer, contentType, apiToken)
+
+	if err != nil {
+		return api.DynatraceEntity{}, err
```

Comment:
I didn't thing about that. Good catch :+1:

Reply:
After thinking about it again, this is not an issue. In the event an `err` is returned
here, it would mean that we didn't even get that far to send a request to the dynatrace
api. 

The case you describe is still handled, because there we get a valid response from
the api, containing an error status code.

Reply:
Nice! Thanks for clarifying. True, we do validate only for resp.code 201.